src/overlays/actors/ovl_En_Ma_Yts/z_en_ma_yts.c

```diff
@@ -21,48 +74,519 @@ const ActorInit En_Ma_Yts_InitVars = {
     (ActorFunc)EnMaYts_Update,
     (ActorFunc)EnMaYts_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Ma_Yts_0x80B8D030/func_80B8D030.asm")
+void EnMaYts_UpdateEyes(EnMaYts* this) {
+    if (this->overrideEyeTexIndex != 0) {
+        this->eyeTexIndex = this->overrideEyeTexIndex;
+    } else if (DECR(this->blinkTimer) == 0) {
+        this->eyeTexIndex++;
+        if (this->eyeTexIndex >= 3) {
+            this->blinkTimer = Rand_S16Offset(30, 30);
+            this->eyeTexIndex = 0;
+        }
+    }
+}
+
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
+    { 18, 46, 0, { 0, 0, 0 } },
+};
+
+static CollisionCheckInfoInit2 D_80B8E19C = {
+    0, 0, 0, 0, 0xFF,
+};
+
+struct struct_80B8E1A8 {
+    /* 0x00 */ AnimationHeader* unk_00;
+    /* 0x04 */ s32 unk_04;
+    /* 0x08 */ u8 unk_08;
+    /* 0x0C */ f32 unk_0C;
+};
+
+static struct struct_80B8E1A8 D_80B8E1A8[] = {
+    { &D_06009E58, 0x3F800000, 0x00, 0.0f }, { &D_06009E58, 0x3F800000, 0x00, -6.0f },
+    { &D_06018948, 0x3F800000, 0x02, 0.0f }, { &D_06018948, 0x3F800000, 0x02, -6.0f },
+    { &D_0601B76C, 0x3F800000, 0x00, 0.0f }, { &D_0601B76C, 0x3F800000, 0x00, -6.0f },
+    { &D_06007328, 0x3F800000, 0x00, 0.0f }, { &D_06007328, 0x3F800000, 0x00, -6.0f },
+    { &D_06014088, 0x3F800000, 0x00, 0.0f }, { &D_06014088, 0x3F800000, 0x00, -6.0f },
+    { &D_06002A8C, 0x3F800000, 0x00, 0.0f }, { &D_06002A8C, 0x3F800000, 0x00, -6.0f },
+    { &D_06015B7C, 0x3F800000, 0x00, 0.0f }, { &D_06015B7C, 0x3F800000, 0x00, -6.0f },
+    { &D_06007D98, 0x3F800000, 0x00, 0.0f }, { &D_06007D98, 0x3F800000, 0x00, -6.0f },
+    { &D_0600852C, 0x3F800000, 0x00, 0.0f }, { &D_0600852C, 0x3F800000, 0x00, -6.0f },
+    { &D_06008F6C, 0x3F800000, 0x00, 0.0f }, { &D_06008F6C, 0x3F800000, 0x00, -6.0f },
+    { &D_060180DC, 0x3F800000, 0x02, 0.0f }, { &D_060180DC, 0x3F800000, 0x02, -6.0f },
```

Comment:
```suggestion
static struct_80B8E1A8 sAnimations[] = {
    { &D_06009E58, 1.0f, 0, 0.0f }, { &D_06009E58, 1.0f, 0, -6.0f },
    { &D_06018948, 1.0f, 2, 0.0f }, { &D_06018948, 1.0f, 2, -6.0f },
    { &D_0601B76C, 1.0f, 0, 0.0f }, { &D_0601B76C, 1.0f, 0, -6.0f },
    { &D_06007328, 1.0f, 0, 0.0f }, { &D_06007328, 1.0f, 0, -6.0f },
    { &D_06014088, 1.0f, 0, 0.0f }, { &D_06014088, 1.0f, 0, -6.0f },
    { &D_06002A8C, 1.0f, 0, 0.0f }, { &D_06002A8C, 1.0f, 0, -6.0f },
    { &D_06015B7C, 1.0f, 0, 0.0f }, { &D_06015B7C, 1.0f, 0, -6.0f },
    { &D_06007D98, 1.0f, 0, 0.0f }, { &D_06007D98, 1.0f, 0, -6.0f },
    { &D_0600852C, 1.0f, 0, 0.0f }, { &D_0600852C, 1.0f, 0, -6.0f },
    { &D_06008F6C, 1.0f, 0, 0.0f }, { &D_06008F6C, 1.0f, 0, -6.0f },
    { &D_060180DC, 1.0f, 2, 0.0f }, { &D_060180DC, 1.0f, 2, -6.0f },
```
Not sure what the convention is for having `0x` in front of small u8 numbers. I think it looks cleaner without it, but it's probably better to follow whatever the convention is. Also, it should also work without the `struct` keyword right?

Reply:
To be able to drop the `struct` keyword it would need to be `typedef`ed.